modules/ROOT/examples/java/src/com/couchbase/code_snippets/Examples.java

```diff
@@ -139,132 +136,531 @@ public static void main (String [] args) throws CouchbaseLiteException, Interrup
-        response) throws javax.servlet.ServletException, IOException {
-            outputMessage("Servlet started :: doPost Invoked");
-            String url = "/showDbItems.jsp";
-            try {
-                MYRESULTS = testCouchbase();
-            } catch (CouchbaseLiteException e) {
-                e.printStackTrace();
-            } finally {
-                outputMessage(String.format("Servlet ended :: There are %d rows in DB", NUMROWS));
-            }
-            request.setAttribute("myRowCount", NUMROWS);
-            request.setAttribute("myResults", MYRESULTS);
-            getServletContext()
-                    .getRequestDispatcher(url)
-                    .forward(request, response);
-            outputMessage("Servlet Ended :: doPost Exits");
+    protected void doPost (javax.servlet.http.HttpServletRequest request, javax.servlet.http.HttpServletResponse
+            response) throws javax.servlet.ServletException, IOException {
+        NUMROWS=0;
+        MYRESULTS="";
+        outputMessage("Servlet started :: doPost Invoked");
+        String url = "/showDbItems.jsp";
+        try {
+            MYRESULTS = testCouchbaseLite();
+        } catch (CouchbaseLiteException | URISyntaxException | InterruptedException e) {
+            e.printStackTrace();
+        } finally {
+            outputMessage(String.format("CouchbaseLite Test Ended :: There are %d rows in DB", NUMROWS));
         }
+        request.setAttribute("myRowCount", NUMROWS);
+        request.setAttribute("myResults", MYRESULTS);
+        getServletContext()
+                .getRequestDispatcher(url)
+                .forward(request, response);
+        outputMessage("Servlet Ended :: doPost Exits");
+    }
 
-    public String testCouchbase() throws CouchbaseLiteException {
+    public String testCouchbaseLite() throws CouchbaseLiteException, URISyntaxException, InterruptedException {
 
 //       public static void main(String[] args) throws Exception {
 
         int randPtr = RANDOM.nextInt(5) +  1;
+        long syncTotal = 0;
         Double randVn = RANDOM.nextDouble() + 1;
-        List<String> listLangs = new ArrayList<String>();
-        listLangs = Arrays.asList("Java","Swift","C#.Net","Objective-C","C++","Cobol");
-        String Prop_Id ="Id";
+        List<String> listLangs = new ArrayList<String> (Arrays.asList("Java","Swift","C#.Net","Objective-C","C++","Cobol"));
+        List<String> listTypes = new ArrayList<String> (Arrays.asList("SDK","API","Framework","Methodology","Language","IDE"));
+
+        String Prop_Id ="id";
         String Prop_Language = "language";
         String Prop_Type = "type";
         String Prop_Version = "version";
+        String searchStringType="SDK";
 
         // Initialise Couchbase
-        outputMessage("== Opening DB and adding record");
 
+//      Initiate CouchbaseLite
         CouchbaseLite.init();
```

Comment:
Normally we would recommend to call init only once when the Web App starts. Calling it multiple time would be fine as well but this means that users will need to remember to do this on every single servlet they have that uses Couchbase Lite. It might be worth to mention about this in the documentation.